src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +70,98 @@ def parse_port_config_file(port_config_file):
+
+        if re.search("\+",breakout_mode) is not None:
+            intf_name_list = []
+            breakout_mode_start_index = re.search(ASYMMETRIC_SPLIT_BREAKOUT_PATTERN, breakout_mode)
+            if breakout_mode_start_index is not None:
+
+                """ first part of port configuration for default breakout mode in Asymmetric form """
+                asymmetric_breakout_mode_start_index, used_lane = int(breakout_mode_start_index.group(1)), int(breakout_mode_start_index.group(4).strip('()'))
+                asymmetric_speed = breakout_mode_start_index.group(2)
+                child_port = used_lane/asymmetric_breakout_mode_start_index
+                start, end = 0, child_port
+                for intf_name in range(index, index+used_lane, child_port):
+                    intf_name_list.append(intf_name)
+                    full_intf_name = "Ethernet"+str(intf_name)
+                    ports[full_intf_name] = {}
+                    ports[full_intf_name]["alias"] = total_alias_list[start]
+                    ports[full_intf_name]["lanes"] = ','.join(total_lanes_list[start:end])
+                    ports[full_intf_name]["speed"] = asymmetric_speed
+                    ports[full_intf_name]["index"] = port_dict[interface_name]['index'][0]
+                    start += child_port
+                    end += child_port
+
+                """ Second part of port configuration for default breakout mode in Asymmetric form """
+                asymmetric_breakout_mode_second_index, second_used_lane = int(breakout_mode_start_index.group(5)), int(breakout_mode_start_index.group(8).strip('()'))
+                asymmetric_speed = breakout_mode_start_index.group(6)
+                child_port = second_used_lane/asymmetric_breakout_mode_second_index
+                index = index + used_lane
+                start, end = used_lane, child_port
+                for intf_name in range(index, index+second_used_lane,child_port):
+                    full_intf_name = "Ethernet"+str(intf_name)
+                    ports[full_intf_name] = {}
+                    ports[full_intf_name]["alias"] = total_alias_list[start]
+                    ports[full_intf_name]["lanes"] = ','.join(total_lanes_list[start:start+end])
+                    ports[full_intf_name]["speed"] = asymmetric_speed
+                    ports[full_intf_name]["index"] = port_dict[interface_name]['index'][0]
+                    start += child_port
+                    end += child_port
+        else:
+            breakout_mode_start_index = re.search(SYMMETRIC_SPLIT_BREAKOUT_PATTERN, breakout_mode)
+            if breakout_mode_start_index is not None:
+
+                """ Port configuration for default breakout mode in symmetric form """
+                symmetric_breakout_mode_start_index = breakout_mode_start_index.group(1)
+                symmetric_speed = breakout_mode_start_index.group(2)
+                child_port = int(LANES)/int(symmetric_breakout_mode_start_index)
```

Comment:
LANES should be the len(lanes_list), not hardcoded.

Reply:
updated as requested. 👍